src/Mappers/BaseMapper.php

```diff
@@ -62,17 +64,158 @@ protected static function getValueFromTag(\DOMDocument $document, string $tag):
         return $element->textContent;
     }
 
-    protected static function getField(\DOMElement $element, string $fieldTagName): ?string
+    protected static function getField(\DOMElement $element, string $fieldTagName, $object = null): ?string
     {
         $fieldElement = $element->getElementsByTagName($fieldTagName)->item(0);
+
         if (!isset($fieldElement)) {
             return null;
         }
 
+        if (isset($object)) {
+            self::checkForMessage($object, $fieldElement);
+        }
+
         if ($fieldElement->textContent === "") {
             return null;
         }
 
         return $fieldElement->textContent;
     }
+    
+    protected static function checkForMessage($object, \DOMElement $element): void
+    {
+        if ($element->hasAttribute('msg')) {
+            $message = new Message();
+            $message->setType($element->getAttribute('msgtype'));
+            $message->setMessage($element->getAttribute('msg'));
+            $message->setField($element->nodeName);
+
+            $object->addMessage($message);
+        }
+    }
+
+    protected static function getAttribute(\DOMElement $element, string $fieldTagName, string $attributeName): ?string
+    {
+        $fieldElement = $element->getElementsByTagName($fieldTagName)->item(0);
+
+        if (!isset($fieldElement)) {
+            return null;
+        }
+
+        if ($fieldElement->getAttribute($attributeName) === "") {
+            return null;
+        }
+
+        return $fieldElement->getAttribute($attributeName);
+    }
+
+    protected static function parseBooleanAttribute(?string $value): ?bool
```

Comment:
Nice, but there is a method in `Util` that does this already?

Reply:
True, but I like this one better as it uses built in PHP functionality and is more versatile. Could probably be moved to `Util`, but functionality is not 1 on 1 comparable and could then have the potential of breaking something.

Reply:
I'm not entirely convinced - I think that it for new functionality it would be hard to decide which boolean parsing function should be used. The existence of two is confusing if there is no clear guideline to pick one over the other.

Please consolidate into one method, its fine to change the one in Util, that's why we have tests.

Reply:
Moved new method to Util